pkg/apis/psmdb/v1/psmdb_types.go

```diff
@@ -626,3 +615,39 @@ func (cr *PerconaServerMongoDB) CanBackup() error {
 
 	return nil
 }
+
+func (s *PerconaServerMongoDBStatus) ClusterStatus(reconcileStatus AppState) AppState {
+	readyRepls := 0
+	for _, replStatus := range s.Replsets {
+		if replStatus.Status == AppStateReady {
+			readyRepls++
+		}
+	}
+
+	switch {
+	case readyRepls == len(s.Replsets) && reconcileStatus == AppStateReady && s.Mongos.Status == AppStateReady:
+		return AppStateReady
+	default:
+		return AppStateInit
+	}
+}
```

Comment:
This method looks a bit odd. Despite its name, it basically checks if a cluster is ready or still initialising. Moreover, there is a similar check for replsets readiness in `updateStatus()` method. And btw the result of that check isn't used.